### ISSUE TITLE
[BUG] twinx()-based plots only generates blank/empty plots

### ISSUE THREAD

**Issue by tai** [JuliaPlots/Plots.jl#3621]
## Details
Plotting dual-axis plot with twinx() results to blank/empty plot.
As you can see in screenshot, x/y1/y2 axis and gridline are drawn, but actual graph is missing.

![image](https://user-images.githubusercontent.com/67324/124554285-c5226080-de70-11eb-9ad8-1957bb1b3a9c.png)

Following minimal example reproduces above results:
```
using Plots
gr()
plot(rand(10))
plot!(twinx(), rand(10))
```

Also, the same issue is reported in Julia discourse: https://discourse.julialang.org/t/plotting-a-cumulative-frequency-line-on-a-histogram-with-second-y-axis/63675/3

This post is quite new, and I believe this feature has been working quite some time now.
So it seems this issue is recently introduced.

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |  x   |     |
pyplot       |  x   |     |
plotly       |     |     | x
plotlyjs     |     |     | x
pgfplotsx    |     |     | x
inspectdr    |     |     | x

### Versions

Plots.jl version: v1.16.6
Backend  version (`]st -m <backend(s)>`):
```
  [28b8d3ca] GR v0.57.5
  [91a5bcdd] Plots v1.16.6
  [d330b81b] PyPlot v2.9.0
```
Output of `versioninfo()`:
```
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2698 v4 @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, broadwell)
```

**Comment by BeastyBlacksmith** [comment]
last version this worked is 1.16.5
```julia
# 1.16.5
julia> using Plots; plot(rand(10)); @which plot!(twinx(), rand(10))
plot!(sp::Plots.Subplot, args...; kw...) in Plots at /home/schrist/.julia/packages/Plots/FKcum/src/plot.jl:217

#1.16.6
julia> using Plots; plot(rand(10)); @which plot!(twinx(), rand(10))
plot!(sp::Plots.Subplot, args...; kw...) in Plots at /home/schrist/.julia/packages/Plots/YVapH/src/plot.jl:249
```

**Comment by BeastyBlacksmith** [comment]
Thats already fixed since 1.17.0
